replica/table.cc

```diff
@@ -1838,6 +1840,33 @@ void table::update_effective_replication_map(locator::effective_replication_map_
     if (old_erm) {
         old_erm->invalidate();
     }
+
+    _stats.tablet_count_on_current_shard = calculate_tablet_count_on_current_shard();
```

Comment:
That would be executed once and will never be updated

Reply:
Okay, so it seems, that I misunderstood Tomek's comment.

> **It could publish table::_tablet_count, which would be set in table::update_effective_replication_map(), which is the place through which all tablet map changes are published.**

Should the calculation be registered via lambda in `table::set_metrics()` then? Would it be okay?

Ref:
https://github.com/scylladb/scylladb/issues/16131#issuecomment-1915564252

Reply:
> Should the calculation be registered via lambda in `table::set_metrics()` then? Would it be okay?

No, calculating in update_effective_replication_map() is better, it's more rare than metric scrape period.

Reply:
The calculation in `update_effective_replication_map()` has been preserved.